File: src/main/java/com/chaottic/nbt/number/IntTag.java
package com.chaottic.nbt.number;

import java.io.DataInput;
import java.io.IOException;

public record IntTag(int i) implements NumberTag {
    @Override
    public byte getAsByte() {
        return (byte) i;
    }

    @Override
    public short getAsShort() {
        return (short) i;
    }

    @Override
    public int getAsInt() {
        return i;
    }

    @Override
    public long getAsLong() {
        return i;
    }

    @Override
    public float getAsFloat() {
        return i;
    }

    @Override
    public double getAsDouble() {
        return i;
    }

    public static IntTag createIntTag(DataInput dataInput) throws IOException {
        return new IntTag(dataInput.readInt());
    }
}

File: src/main/kotlin/com/chaottic/nbt/number/IntTag.kt
package com.chaottic.nbt.number

import com.chaottic.nbt.TagType
import java.io.DataInput

@JvmInline
value class IntTag(private val i: Int) : NumberTag {

	override fun asByte() = i.toByte()
	override fun asShort() = i.toShort()
	override fun asInt() = i
	override fun asLong() = i.toLong()
	override fun asFloat() = i.toFloat()
	override fun asDouble() = i.toDouble()

	companion object : TagType<IntTag> {

		override fun createTag(dataInput: DataInput) = IntTag(dataInput.readInt())
	}
}
